### ISSUE TITLE
Vue Router

### ISSUE THREAD

**Issue by olimart** [olimart/kanban#9]
For now, server returns all boards at once.

Need to implement [Vue router](https://github.com/vuejs/vue-router) when switching from one board to another to fetch data from server.

When requesting data from server we should use [Vue resource](https://github.com/vuejs/vue-resource)

**Comment by kyrsquir** [comment]
What are API calls for separate boards?

**Comment by olimart** [comment]
/api/boards/:slug

**Comment by kyrsquir** [comment]
How do I know what is the slug of the needed board?
It seems I first need to get the list of all existing slugs and corresponding boards' names (to populate board selection menu).
Also should I request for active board's containment on every switch between boards or only for the first one?

**Comment by olimart** [comment]
> How do I know what slug the needed board has? It seems I first need to get the list of all existing slugs and corresponding boards' names (to populate board selection menu).

For now use the `/api/boards` call (which is current loading behaviour). In future I might create a lighter API calls such as `/api/boards/sidebar` but not ready for that yet.
Make sure it persists when you switch to a different board (so no additional call when you switch to a different board).

> Also should I request for active board's containment on every switch between boards or only for the first time I switch to it?

every time you switch to board

**Comment by kyrsquir** [comment]
For now I implemented board routing without the Vue Router.
Could you please explain, what is it (Vue Router) for? I failed to understand its purpose.

Also I've restructured the app into components so there may appear some new bugs, I will test it more thoroughly after the migration to 1.0.

**Comment by olimart** [comment]
yeah, vue router may not be the right term and not appropriate for switching board.

My point is that external calls (call to the backend API) should be done with `vue-resource`

**Comment by kyrsquir** [comment]
While testing I've noticed a problem. With the current way of routing (sending the whole board upon every update), when opening a card/lists/comments form, closing all opened forms of the same type isn't enough. Data is still getting overwritten unless ALL forms related to current board are closed which sounds like a performance-heavy task.

Do you still want me to look for all components with unsaved data and close them? Or will the routing api be changed at some point?

**Comment by olimart** [comment]
I'm planning to separate boards, lists and cards in near future.
